src/main.rs

```diff
@@ -399,6 +396,10 @@ fn notmain() -> Result<i32, anyhow::Error> {
         .as_ref()
         .map_or(false, |ch| ch.name() == Some("defmt"));
 
+    if use_defmt && opts.no_flash {
+        bail!("\"defmt\" RTT channel is in use, but `--no-flash` was specified -- this combination is not allowed");
```

Comment:
nit: `"defmt" RTT channel is in use` requires system knowledge in order to grasp what the problem is– could this be rephrased to something like "attempting to specify `--no-flash`while using `defmt` -- this combination is not allowed"?